### PR TITLE
Fix "No module named 'pattern_library'" ImportError

### PR DESCRIPTION
**Pull request by jhonatan-lopes**
Related: #9665 #10808 #10898

# Description

Fixes `ImportError: No moduel named 'pattern_library'` when building the web app.

Modifies the settings to only import the `project_styleguide` app if the pattern library is enabled.

Moves `django-pattern-library` dependency to `requirements.in` instead of `dev-requirements.in`.

### CODE REVIEW

File: network-api/networkapi/settings.py [hunk 1]
```diff
@@ -257,6 +262,8 @@
             "networkapi.wagtailpages",
             "networkapi.mozfest",
             "networkapi.donate",
+            "pattern_library" if PATTERN_LIBRARY_ENABLED else None,
+            "networkapi.project_styleguide" if PATTERN_LIBRARY_ENABLED else None,
```
Comment (jhonatan-lopes) [inline network-api/networkapi/settings.py:266]: This is the only change in this PR besides re-enabling the pattern library.

**Comment by jhonatan-lopes** [comment]
@mmmavis and @mtdenton it seems that the issue was that `django-pattern-library` was installed as a dev-dependency when it should be a regular dependency, since it is imported in the code. I've tried deploying a review app and the build was successful but I got an error somewhere in the post-build scripts:

```
Creating Donate Site record in Wagtail

[32;1mDone![0m

Setting up DNS records and Sites for review app

Traceback (most recent call last):

  File "/app/network-api/networkapi/management/commands/review_app_admin.py", line 20, in handle

    User.objects.get(username="admin")

  File "/app/.heroku/python/lib/python3.9/site-packages/django/db/models/manager.py", line 85, in manager_method

    return getattr(self.get_queryset(), name)(*args, **kwargs)

  File "/app/.heroku/python/lib/python3.9/site-packages/django/db/models/query.py", line 435, in get

    raise self.model.DoesNotExist(

django.contrib.auth.models.User.DoesNotExist: User matching query does not exist.



During handling of the above exception, another exception occurred:



Traceback (most recent call last):

  File "/app/network-api/manage.py", line 35, in <module>

    execute_from_command_line(sys.argv)

  File "/app/.heroku/python/lib/python3.9/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line

    utility.execute()

  File "/app/.heroku/python/lib/python3.9/site-packages/django/core/management/__init__.py", line 413, in execute

    self.fetch_command(subcommand).run_from_argv(self.argv)

  File "/app/.heroku/python/lib/python3.9/site-packages/django/core/management/base.py", line 354, in run_from_argv

    self.execute(*args, **cmd_options)

  File "/app/.heroku/python/lib/python3.9/site-packages/django/core/management/base.py", line 398, in execute

    output = self.handle(*args, **options)

  File "/app/network-api/networkapi/management/commands/review_app_admin.py", line 47, in handle

    r.raise_for_status()

  File "/app/.heroku/python/lib/python3.9/site-packages/requests/models.py", line 1021, in raise_for_status

    raise HTTPError(http_error_msg, response=self)

requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: https://api.github.com/repos/mozilla/foundation.mozilla.org/pulls/10930
```
although I do think this might be related to something else.

**Comment by mtdenton** [comment]
Yeah IIRC @mmmavis has pointed out that error before concerning our review apps.

**Comment by mmmavis** [comment]
I updated the PR description to include the GitHub PRs and tickets this PR is related to . Please have a look and correct them if there's any mistake :)

**Review by mmmavis — APPROVED** [review]
✨

**Comment by mmmavis** [comment]
@mtdenton @jhonatan-lopes I did a search for `requests.exceptions.HTTPError:` on Slack and it looks like this error has been around for a while 😬

**Pull request merged**